src/metalperformanceshaders.cs

```diff
@@ -5877,6 +6007,154 @@ interface MPSKeyedUnarchiver : MPSDeviceProvider {
 		//IntPtr Constructor (NSData data, IMTLDevice device);
 	}
 
+	[Introduced (PlatformName.MacCatalyst, 13, 0)]
+	[TV (13,0), Mac (10,15), iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface MPSNDArrayDescriptor
+	{
+		[Export ("dataType", ArgumentSemantic.Assign)]
+		MPSDataType DataType { get; set; }
+
+		[Export ("numberOfDimensions")]
+		nuint NumberOfDimensions { get; set; }
+
+		[Export ("lengthOfDimension:")]
+		nuint GetLength (nuint dimensionIndex);
+
+		[Export ("sliceRangeForDimension:")]
+		MPSDimensionSlice GetSliceRange (nuint dimensionIndex);
+
+		[Export ("sliceDimension:withSubrange:")]
+		void Slice (nuint dimensionIndex, MPSDimensionSlice subRange);
+
+		[Export ("transposeDimension:withDimension:")]
+		void Transpose (nuint dimensionIndex, nuint dimensionIndex2);
+
+		// -(vector_uchar16)dimensionOrder;
+		// [Export ("dimensionOrder")]
+		// [Verify (MethodToProperty)]
+		// [unsupported ExtVector: unsigned char __attribute__((ext_vector_type(16)))] DimensionOrder { get; }
+
+		[Static]
+		[Export ("descriptorWithDataType:dimensionCount:dimensionSizes:")]
+		MPSNDArrayDescriptor Create (MPSDataType dataType, nuint numberOfDimensions, IntPtr dimensionSizes);
+
+		[Static]
+		[Export ("descriptorWithDataType:shape:")]
+		MPSNDArrayDescriptor Create (MPSDataType dataType, NSNumber[] shape);
+
+		// This is implemented using params and descriptorWithDataType:dimensionCount:dimensionSizes: in MPSNDArray.cs
+		// [Static, Internal]
+		// [Export ("descriptorWithDataType:dimensionSizes:", IsVariadic = true)]
+		// MPSNDArrayDescriptor Create (MPSDataType dataType, nuint dimension0, IntPtr varArgs);
+
+		[Export ("reshapeWithDimensionCount:dimensionSizes:")]
+		void Reshape (nuint numberOfDimensions, IntPtr dimensionSizes);
+
+		[Export ("reshapeWithShape:")]
+		void Reshape (NSNumber[] shape);
```

Comment:
could this be a more concrete type? do you happen to know? if you do then you can do something like this

```csharp
		void Reshape ([BindAs (typeof (nint []))] NSNumber[] shape);
```

BindAs docs https://docs.microsoft.com/en-us/xamarin/cross-platform/macios/binding/objective-c-libraries?tabs=macos#binding-nsvalue-nsnumber-and-nsstring-to-a-better-type

Reply:
Ah, no I didn't know that, thanks!

Thing is, I *really* want `params` for these puppies and that's why I provided the `params` version in manual code. Is there an attribute to get `params`?